relengapi/blueprints/slaveloan/bugzilla.py

```diff
@@ -0,0 +1,142 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# XXX Much of this functionality should probably be its own
+#     Separate relengapi blueprint
+
+import logging
+
+from bzrest.client import BugzillaClient
+from flask import current_app
+from flask import url_for
+
+log = logging.getLogger(__name__)
+
+MAX_ALIAS = 15
+DEFAULT_BUGZILLA_URL = "https://bugzilla-dev.allizom.org/rest/"
+
+
+def _bzclient():
+    if getattr(current_app, 'bzclient', None):
+        return current_app.bzclient
+    # XXX: Todo log warning if user or pass is "None"
+    bzclient = BugzillaClient()
+    bzclient.configure(bzurl=current_app.config.get('BUGZILLA_URL', DEFAULT_BUGZILLA_URL),
+                       username=current_app.config.get('BUGZILLA_USER', None),
+                       password=current_app.config.get('BUGZILLA_PASS', None))
+    current_app.bzclient = bzclient
+    return current_app.bzclient
+
+
+class Bug(object):
+    alias = None
+    id = None
+
+    def __init__(self, id_=None, loadInfo=True):
+        if isinstance(id_, int):
+            self.id = id_
+        else:
+            self.alias = id_
+        self.data = {}
+        if id_ and loadInfo:
+            self.refresh()
+
+    @property
+    def id_(self):
+        return self.id or self.alias
+
+    def refresh(self):
+        self.data = _bzclient().get_bug(self.id_)
+        self.id = self.data["id"]
+        self.alias = self.data["alias"]
+
+    def add_comment(self, comment, data={}):
+        return _bzclient().add_comment(self.id_, comment, data)
+
+
+class ProblemTrackingBug(Bug):
+    product = "Release Engineering"
+    component = "Buildduty"
+
+    def __init__(self, slave_name, *args, **kwargs):
+        self.slave_name = slave_name
+        Bug.__init__(self, id_=slave_name, *args, **kwargs)
+
+    def create(self, comment=None, depends_on=None):
+        if len(self.slave_name) > MAX_ALIAS:
+            alias = None
+        else:
+            alias = self.slave_name
+        data = {
+            "product": self.product,
+            "component": self.component,
+            "summary": "%s problem tracking" % self.slave_name,
+            "version": "other",
+            "alias": alias,
+            # todo: do we care about setting these correctly?
+            "op_sys": "All",
+            "platform": "All"
+        }
+        if comment:
+            data['comment'] = comment
+        if depends_on:
+            data['depends_on'] = depends_on
+        resp = _bzclient().create_bug(data)
+        self.id = resp["id"]
+
+
+class LoanerBug(Bug):
```

Comment:
You use "loaner bug" here but "lone bug" below - might be confusing for other devs trying to remember which is named which